doc/development/adding_operators.rst

```diff
@@ -0,0 +1,357 @@
+        # In our case this may be one or two, which is why we
+        # use the AnyWires Enumeration to indicate a variable number.
+        num_wires = qml.operation.AnyWires
+
+        # This attribute tells PennyLane what differentiation method to use. Here
+        # we request parameter-shift (or "automatic") differentiation.
+        grad_method = "A"
+
+        def __init__(self, angle, wire_rot, wire_flip=None, do_flip=False,
+                           do_queue=True, id=None):
+
+            # checking the inputs --------------
+
+            if do_flip and wire_flip is None:
+                raise ValueError("Expected a wire to flip; got None.")
+
+            # note: we use the framework-agnostic math library since
+            # trainable inputs could be tensors of different types
+            shape = qml.math.shape(angle)
+            if len(shape) > 1:
+                raise ValueError(f"Expected a scalar angle; got angle of shape {shape}.")
+
+            #------------------------------------
+
+            # do_flip is not trainable but influences the action of the operator,
+            # which is why we define it to be a hyperparameter
+            self._hyperparameters = {
+                "do_flip": do_flip
+            }
+
+            # we extract all wires that the operator acts on,
+            # relying on the Wire class arithmetic
+            all_wires = qml.wires.Wires(wire_rot) + qml.wires.Wires(wire_flip)
+
+            # The parent class expects all trainable parameters to be fed as positional
+            # arguments, and all wires acted on fed as a keyword argument.
+            # The id keyword argument allows users to give their instance a custom name.
+            # The do_queue keyword argument specifies whether or not
+            # the operator is queued when created in a tape context.
+            super().__init__(angle, wires=all_wires, do_queue=do_queue, id=id)
+
+        @property
+        def num_params(self):
+            # if it is known before creation, define the number of parameters to expect here,
+            # which makes sure an error is raised if the wrong number was passed
+            return 1
+
+        @staticmethod
+        def compute_decomposition(angle, wires, do_flip):  # pylint: disable=arguments-differ
+            # Overwriting this method defines the decomposition of the new gate, as it is
+            # called by Operator.decomposition().
+            # The general signature of this function is (*parameters, wires, **hyperparameters).
+            op_list = []
+            if do_flip:
+                op_list.append(qml.PauliX(wires=wires[1]))
+            op_list.append(qml.RX(angle, wires=wires[0]))
+            return op_list
+
+        def adjoint(self):
+            # the adjoint operator of this gate simply negates the angle
+            return FlipAndRotate(-self.parameters[0], self.wires[0], self.wires[1], do_flip=self.hyperparameters["do_flip"])
```

Comment:
This is nice, I think it now makes sense in this example why you would subclass rather than just write a qfunc :) Because here you are defining more than one representation